File: c++/dict.cpp
#include<iostream>
#include<map>
#include<string>
using namespace std;

int main(){
    map<string,int> dict;
    dict["a"]=1;
    dict["aa"]=2;
    dict["abcd"]=1234;

    cout<<"dict[\"a\"] : "<<dict["a"]<<endl;
    cout<<"dict[\"aa\"] : "<<dict["aa"]<<endl;
    cout<<"dict[\"abcd\"] : "<<dict["abcd"]<<endl;
    cout<<"dict[\"a\"]+dict[\"aa\"] : "<<dict["a"]+dict["aa"]<<endl;
    dict["a"]++;
    cout<<"dict[\"a\"]++ : "<<dict["a"]<<endl;
    dict["a"]=3;
    cout<<"dict[\"a\"] : "<<dict["a"]<<endl;
    cout<<"dict[\"b\"] : "<<dict["b"]<<endl;

    cout<<"using dict.count()"<<endl;
    cout<<"dict.count(\"a\") : "<<dict.count("a")<<endl;
    cout<<"dict.count(\"b\") : "<<dict.count("b")<<endl;
    cout<<"dict.count(\"z\") : "<<dict.count("z")<<endl;
    cout<<"dict.count(\"z\") : "<<dict.count("z")<<endl;

    int val_to_find=1234;
    string key_present;
    bool found=false;
    for(auto pair:dict){
        if(pair.second==val_to_find){
            found=true;
            key_present=pair.first;
            break;
        }
    }
    if(found){
        cout<<"value set for key : "<<key_present<<endl;
    }
    else{
        cout<<"Not present in dictionary"<<endl;
    }

    map<int,char> dict2;
    dict2[0]='a';
    dict2[1]='b';
    auto it=dict2.find(1); // .find(key)
    cout<<"key : "<<it->first<<" value : "<<it->second<<endl;
}

File: PYTHON/dict.py

dict = {
        "a": 1,
        "aa": 2,
        "abcd": 1234
}
    
print(f'dict["a"] : {dict["a"]}')
print(f'dict["aa"] : {dict["aa"]}')
print(f'dict["abcd"] : {dict["abcd"]}')
print(f'dict["a"] + dict["aa"] : {dict["a"] + dict["aa"]}')
    
dict["a"] += 1
print(f'dict["a"]+=1 : {dict["a"]}')
    
dict["a"] = 3
print(f'dict["a"] : {dict["a"]}')
print(f'dict.get("b", 0) : {dict.get("b", 0)}')

print("using dict.get()")
print(f'dict.get("a", 0) : {dict.get("a", 0)}')
print(f'dict.get("b", 0) : {dict.get("b", 0)}')
print(f'dict.get("z", 0) : {dict.get("z", 0)}')

val_to_find = 1234
key_present = None
found = False
for key, value in dict.items():
    if value == val_to_find:
        found = True
        key_present = key
        break
    
if found:
    print(f'value set for key : {key_present}')
else:
    print("Not present in dictionary")

dict2 = {0: 'a', 1: 'b'}
it = dict2.get(1)
if it is not None:
    print(f'key : 1 value : {it}')
print("dict values : ",dict)
dict.pop("abcd") # key
del dict["aa"] # value
print("dict values after pop() and del : ",dict)

# to check if a key is present
if "a" in dict :
    print("Key \"a\" is present")
else :
    print("Key \"a\" is not present")